src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public class zip_LargeFiles : ZipFileTestBase
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized))] // don't run it on slower runtimes
+        [OuterLoop("It takes more than 10 minutes")]
+        public static void UnzipOver4GBZipFile()
+        {
+            Random random = new (12345); // use const seed for reproducible results
+            byte[] buffer = new byte[1_000_000]; // 1 MB
+            Dictionary<string, byte[]> entiresContent = new();
+
+            string zipArchivePath = Path.Combine("ZipTestData", "over4GB.zip");
+            DirectoryInfo tempDir = Directory.CreateDirectory(Path.Combine("ZipTestData", "over4GB"));
+
+            try
+            {
+                for (int i = 0; i < 4_500; i++)
+                {
+                    random.NextBytes(buffer);
+
+                    string entryName = $"{i}.test";
+                    File.WriteAllBytes(Path.Combine(tempDir.FullName, entryName), buffer);
+                    entiresContent.Add(entryName, buffer.ToArray());
+                }
+
+                ZipFile.CreateFromDirectory(tempDir.FullName, zipArchivePath);
```

Comment:
Maybe use CompressionLevel.Fastest or NoCompression to save time?

Reply:
> Maybe use CompressionLevel.Fastest or NoCompression to save time?

@iSazonov thank you, it's a good suggestion!

I've tried it and for `CompressionLevel.NoCompression` it takes 10 minutes to run the test, while for `CompressionLevel.Fastest` it's 12 minutes, almost identical to current implementation (`CompressionLevel.Optimal` is the default).

Since there is no big difference between 10 minutes and 12 and `CompressionLevel.Optimal` is closer to real life scenarios I am going to keep it as it is for now (I am also going to experiment with fewer, but larger files).

Reply:
I guess `random.NextBytes` takes ~10%.
So CI disk is too slow. Maybe use a RAM disk if possible?